src/pages/workspaces/workspace/delete-workspace-modal.state.ts

```diff
@@ -0,0 +1,111 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { Ajax } from 'src/libs/ajax'
+import { reportError, withErrorReportingInModal } from 'src/libs/error'
+import { useCancellation, useOnMount } from 'src/libs/react-utils'
+import { getUser } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
+import { BaseWorkspace, isGoogleWorkspace, WorkspaceInfo } from 'src/libs/workspace-utils'
+import { isResourceDeletable } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+
+
+interface DeleteWorkspaceModalLeoApp {
+  appName: string
+  status: string
+  cloudContext: any
+}
+
+export interface DeleteWorkspaceState {
+  loading: boolean
+  deleting: boolean
+  isDeleteDisabledFromResources: boolean
+  workspaceBucketUsageInBytes: number | null
+  deletableApps: DeleteWorkspaceModalLeoApp[]
+  nonDeletableApps: DeleteWorkspaceModalLeoApp[]
+  collaboratorEmails: string[] | null
+  hasApps: () => boolean
+  deleteWorkspace: () => void
+  controlledResourcesExist: boolean
+}
+
+
+export const useDeleteWorkspaceState = (workspace: BaseWorkspace, onDismiss: () => void, onSuccess: () => void) : DeleteWorkspaceState => {
+  const [deleting, setDeleting] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const [apps, setApps] = useState<DeleteWorkspaceModalLeoApp[]>()
+  const [collaboratorEmails, setCollaboratorEmails] = useState<string[] | null>()
+  const [workspaceBucketUsageInBytes, setWorkspaceBucketUsageInBytes] = useState<number | null>()
+  const [controlledResourcesExist, setControlledResourcesExist] = useState(false)
+  const [deletableApps, nonDeletableApps] = _.partition(isResourceDeletable('app'), apps) as [DeleteWorkspaceModalLeoApp[], DeleteWorkspaceModalLeoApp[]]
+
+  const workspaceInfo: WorkspaceInfo = workspace.workspace
+  const signal = useCancellation()
+
+  useOnMount(() => {
```

Comment:
Most of the logic in this hook was lifted straight from the original `DeleteWorkspaceModal`, with some type changes.